python/tk_multi_publish2/api/tree.py

```diff
@@ -0,0 +1,284 @@
+# Copyright (c) 2018 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import traceback
+
+import json
+
+import sgtk
+from .item import PublishItem
+
+logger = sgtk.platform.get_logger(__name__)
+
+
+class PublishTree(object):
```

Comment:
again, this is a public API now so have a look through to see if seems reasonable.